universe/NamedValueRefManager.h

```diff
@@ -79,8 +79,23 @@ struct FO_COMMON_API NamedRef final : public ValueRef<T>
     }
 
     std::string Dump(unsigned short ntabs = 0) const override {
-        auto ref = GetValueRef();
-        return ref ? ref->Dump() : "NAMED_REF_UNKNOWN";
+        std::string retval = "Named";
+        if constexpr (std::is_same<T, int>::value) {
+            retval += "Integer";
+        } else if constexpr (std::is_same<T, double>::value) {
+            retval += "Real";
+        } else {
+            retval += typeid(T).name();
```

Comment:
This one is tricky on multiple accounts. First typeid(T).name is implementation defined/unspecified and might do weird stuff.
Also AFAICR the Dump function should look like the FOCS specification. Looking at the parser one can do Namedplanettype and Namedenvironment in FOCS (I merely added it for proof of concept). You ~could check for those types I guess(?)

Reply:
Else I personally would not mind if you return "Generic" here.

Reply:
Used `Generic` here.